### ISSUE TITLE
New scanner: features (scan-features:) [PAXRUNNER-227]

### ISSUE THREAD

**Issue by ops4j-issues** [ops4j/org.ops4j.pax.runner#258]
**[Alin Dreghiciu](https://ops4j1.jira.com/secure/ViewProfile.jspa?accountId=557058:a786db11-2013-41fb-af1d-ca5eb8f74aa0)** created **[PAXRUNNER-227](https://ops4j1.jira.com/browse/PAXRUNNER-227)** 

ServiceMix4 introduced it's own kind of profiles named features. It would be usefull if Pax Runner will have a scanner for the so called SMX4 features repositories = xml files grouping bundles (urls) into features.

Proposed syntax: scan-features:&lt;feature file url&gt;!/&lt;comma separated list of feature names&gt;<br /> where a feature name can include also a version separated by "/".

E.g. (based on the example of file bellow)<br /> scan-features:mvn:org.apache.servicemix.nmr/apache-servicemix-nmr/1.0.0-m4/xml/features!/nmr,nmr-audit,jbi/1.0.0

Here are some pointers:<br /> [http://servicemix.apache.org/SMX4KNL/66-installing-additional-features.html](http://servicemix.apache.org/SMX4KNL/66-installing-additional-features.html)<br /> Example of feature [file:http://repo2.maven.org/maven2/org/apache/servicemix/nmr/apache-servicemix-nmr/1.0.0/apache-servicemix-nmr-1.0.0-features.xml](file:http://repo2.maven.org/maven2/org/apache/servicemix/nmr/apache-servicemix-nmr/1.0.0/apache-servicemix-nmr-1.0.0-features.xml)

---

**Fixed in:** 0.18.0
Votes: 0, Watches: 0

**Comment by ops4j-issues** [comment]
**[Toni Menzel](https://ops4j1.jira.com/secure/ViewProfile.jspa?accountId=557058:e7e50a9b-1dd6-45df-b446-7b0e65306e2b)** commented

also we should really think about aligning the profile specification formats.

**Comment by ops4j-issues** [comment]
**[Alin Dreghiciu](https://ops4j1.jira.com/secure/ViewProfile.jspa?accountId=557058:a786db11-2013-41fb-af1d-ca5eb8f74aa0)** commented

I did studied the features before I started pax runner profile re-factoring and scan-composite. I think our is much powerful and works well with the other scanner taking advantage of existing resources and formats we have and support.



And by having this scanner we can support anything. If people like this format they can use the scanner with no problem but I would not trade out composite format for this one as we can include any type of resources now with scan-composite as fo example the powerful scan-maven: about to be created.



Last but not least I do not like the xml format.
